code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+/obj/machinery/mind_machine
+	name = "Mind Machine"
+	icon = 'icons/obj/mind_machine.dmi'
+	density = 1
+	anchored = 1.0
+	use_power = 1
+	idle_power_usage = 50
+	active_power_usage = 2000
+	light_range_on = 1
+	light_power_on = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | EMAGGABLE | FIXED2WORK | WRENCHMOVE
+
+/obj/machinery/mind_machine/mind_machine_hub
+	name = "Mind Machine Hub"
+	icon_state = "mind_hub"
+	desc = "The main hub of a complete mind machine setup. Placed between two mind pods and used to control and manage the transfer. Houses an experimental bluespace conduit which uses bluespace crystals for charge."
+	var/bluespaceConduit = 0 //Swap ammo
+	var/soulShardSafety = FALSE	//Day ruin insurance
+	var/obj/machinery/mind_machine/mind_machine_pod/connectOne //All "variableNameOne" refer to information about the connectOne pod
+	var/obj/machinery/mind_machine/mind_machine_pod/connectTwo //Likewise for variableNameTwo for connectTwo
+	var/podsConnected = FALSE
+	var/occupantOne = null
+	var/occupantTwo = null
+	var/occupantNameOne = "None"
+	var/occupantNameTwo = "None"
```

Comment:
ditto

Reply:
See the problem with all these vars down here is that while they *are* completely inferrable from the contained mobs, it's kind of non-trivial I guess?

I would prefer if this data wasn't duplicated and the data was pulled from the mobs in `ui_interact()` instead to reduce duplication though, yeah.

Also makes it more clear what this is used for and the user-intended strings are then located in only `ui_interact()`